### ISSUE TITLE
Make Ctrl+M a global action and not an editor action

### ISSUE THREAD

**Issue by stevencl** [Microsoft/vscode#3285]
I turned tab to move focus on.

I tabbed into the git viewlet, then selected a file with changes. I pressed enter. The diff editor opened up to show the changes. I pressed tab a few times and watched focus move through the editor actions. When focus shifted to the editor, tab inserted a tab in the editor and I noticed that the status bar was no longer showing the message that tab shifts focus.

As soon as I opened another file though, the status bar changed and indicated that tab moves focus is back on.

I expected tab to move focus to stay on when viewing the diff editor as I didn't want to mistakenly edit the file.

Found while testing #3246

**Comment by bpasero** [comment]
@stevencl unfortunately we do not have a global action to enable tab to move focus on, instead you have to enable it per editor. The diff editor is not a normal editor, so when it opens, you have to enable it again.

@alexandrudima I wonder if the action to enable tab navigation should be global for all editors, instead of local to each editor.

**Comment by alexandrudima** [comment]
Not sure either, the accessibility docs ask that a widget that traps focus must allow for a way to escape it. It doesn't specify how this should be implemented

@stevencl Do you know what VS does?

**Comment by stevencl** [comment]
Sorry @alexandrudima, I'm not sure what you are asking?

**Comment by alexandrudima** [comment]
@stevencl Once you focus an editor, pressing <kbd>Tab</kbd> inserts tab characters (i.e. does not move focus). This means the editor is trapping focus, you cannot focus the next element in the tab ring. We currently allow you to escape the trap by toggling what <kbd>Tab</kbd> does by pressing <kbd>Ctrl+M</kbd>. Once you are in an editor and you press <kbd>Ctrl+M</kbd> the specific editor you were in allows you to escape it (<kbd>Tab</kbd> moves focus). However, once you land on another editor, you are trapped again and you must do it again if you want to focus the next focusable element.

If you land on a readonly editor, then focus is not trapped since you cannot edit it.

**Comment by alexandrudima** [comment]
As far as I can tell, this is compliant with the accessibility docs, as it provides a means to escape a focus trap, but the means are local to the current editor. In other words <kbd>Ctrl+M</kbd> is a local toggle and not a global one. The question was if you know how VS handles this?

**Comment by alexandrudima** [comment]
add9eedf305a5487fa3decbf87bb6ae969de97d2 makes `Ctrl+M` a global concept
